From: gregkh@suse.de
Subject: [PATCH] USB: fix memory leak in get_string if usb_string() call failed.

[PATCH] USB: fix memory leak in get_string if usb_string() call failed.

Pointed out by Alan Stern <stern@rowland.harvard.edu>

Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>


 drivers/usb/core/hub.c |    2 ++
 1 files changed, 2 insertions(+)


diff -Nru a/drivers/usb/core/hub.c b/drivers/usb/core/hub.c
--- a/drivers/usb/core/hub.c	2005-03-08 16:40:34 -08:00
+++ b/drivers/usb/core/hub.c	2005-03-08 16:40:34 -08:00
@@ -1125,6 +1125,8 @@
 		return;
 	if (usb_string(udev, index, buf, 256) > 0)
 		*string = buf;
+	else
+		kfree(buf);
 }
 
 
